### ISSUE TITLE
Sprint - May 27 to June 7, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#22405]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by pauldambra** [comment]
# Team A nightmare on CSS parsing street

**Support hero:** @manoel 

* bank holiday monday
* manoel out a few days
* paul out for week 1

## items from previous sprint

### High priority

-  🟡 hiring (@pauldambra)
    - 	handed mobile engineer tech screens to Manoel - it was silly me doing them
    - three more FS engineer tech screensbooked
- hogql filters (@daibhin  with @pauldambra  supporting)
    - this is out for a percentage of traffic
    - it is at least twice as slow
    - person properties are expressed as _two_ joins between events and persons/pdi
    - repeated parsing is ~30% of processing time
- 💚 Replace posthog-java with posthog (The core of the Android SDK written in Kotlin) [issue](https://github.com/PostHog/posthog-java/issues/50). (@marandaneto )
    - on course - Neil and I being mean :p

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

- 🟡 quality and consolidation
    - mobile replay open beta
        - out free and being promoted in onboarding
    - replay captures every site
        - David working with rrweb to improve CSS parsing, already a major reduction in issues as a result
    - people can find the valuable recordings 
        - queued behind the others
- 🟡 Error tracking MVP
    - need to decide if we start
- 🟡 Hire _at least_ 2 amazing colleagues
   
### High priority

- hiring (@pauldambra)
- hogql filters performance (@pauldambra )
    - intercept person property queries and convert them to subquery not join (housewatch thinks 14x faster)
    - can't roll out as is


### Low priority / side quests

- mobile usage report stuff @pauldambra 
 - persistent replay queue in posthog-js @pauldambra 
- Remove last of the antd components @daibhin

**Comment by EDsCODE** [comment]
# Team Data Shack

### OKR Q2 2024

### Objective

**Release data warehouse to everyone**

* **Key Results:**
    * Integration first experience
         * schemas are reliable
         * modeling of each integration is clear
         * Good automatic roll up views and joins
         * Wizard to onboard people
    * Establish a solid pattern to build integrations
    * Complete data warehouse experience in the rest of the app (insights, feature flags, experiments)

**James as a Service -> Clickhouse as a Service**

* **Key Results:**
    * Better Visibility
         * Regularly testing backups
         * Monitoring/alerting
         * Mutations
         * Moves
    * Management
         * Managing/killing mutations
    * Self Serve
         * Schema design feedback (James non blocking
         * Schema management   
    * Automation
         * Replace/Upgrade replicas
         * Upgrading to 24.04
         * Disk configs 
 
### Retro

- offsite

**Product**
- making the UI changes to move sources into pipeline 3000
- Hackathon
- [ ] logic refactors and cleanup @Gilbert09 
- [ ] progressing through https://github.com/PostHog/posthog/issues/22001 @EDsCODE 

**Infra**
- [ ] Hackathon
- [ ] Start EU replication
- [ ] Upgrade EU Cluster to 23.12
- [ ] Increase storage on EU instances

### High priority

**Product**
- [ ] complete https://github.com/PostHog/posthog/issues/22001 (decide on pricing, UI changes, instrumentation) @EDsCODE

**Comment by mariusandra** [comment]
# Team Internal Tools / CDP / Hog / QL

### High priority

- HogQL-related performance fixes
  - Session replay filters
  - Product analytics "person properties into subquery"

- Figure out what to build
   - and in what order
   - and get going.

### Low priority / side quests

- Read about compilers and type systems.

**Comment by benjackwhite** [comment]
# Team Infra

## Hang over items from previous sprint

1.  🟡 CDP v1.5 - Webhooks destination filtered by Action @benjackwhite 
- Just waiting on merges now
3. 🟡  Get as many other log sources into Loki as possible (Kubernetes audit logs, Cloudtrail, SQL/Clickhouse queries, Metabase, HogQL queries) @danielxnj 
- K8s audit logs, cloudtrail ✅ 
- Metabase - just needs a license
- HogQL - needs our own implementation 
4. 🔴 Get more useful auditing logs out of Django (auth attempts, access logs, HogQL queries) @benjackwhite 
5. 🔴 Get Metabase setup to tick the audit boxes (auditing, SSO etc.) @frankh @danielxnj 
6. 🟢 Continue with Soc2 Audit follow up @danielxnj 
7. 🟢 Fix all remaining High/medium pen test outcomes @benjackwhite 
8. 🟢 Offsite!

## OKR

1. 💪 Deploy with confidence 🟢 
- Finalize our Canary Deploy process 🟡
5. 🚨 Improved alerting and monitoring 🟢 
- SIEM work has made this a priority
6. 🔒 Deeper Security 🟡 
9. 💰 Continued cost control 🟢 

### High priority

- Daniel - vacation
- Metabase setup and auditting / SSO / move to internal ELB @frankh 
- Reverse proxy with release @frankh
- Investigate canary deploy rollback improvements @frankh

**Comment by robbie-c** [comment]
# Team Web Analytics

**Support hero:** n/a 

## Retro
Sickonos + reprioritisation meant that I didn't get much done of what was in hindsight quite an ambitious sprint:
- 🟢 finish this spike on uuidv7s
- 🔴 write the SQL for the new sessions table that uses it
- 🔴 write and run a proper backfill job
- 🔴 switch the hogql sessions table over to the new table
- 🔴 make sure docs / sdks make it clear enough that session_ids should be a UUIDv7
- 🔴 then roll out session properties elsewhere in the app
- 🟡  mop up personless events, TBD(mykonos) what this involves

## OKR

1. Make querying fast enough for large customers
2. Do personless events work where necessary (unknown amount of work)
3. Iterate on customer feedback
4. Product management work

### High priority
- test left join changes on persons table 
- write the SQL for the new sessions table that uses it
- write and run a proper backfill job
- switch the hogql sessions table over to the new table
- make sure docs / sdks make it clear enough that session_ids should be a UUIDv7

**Comment by Twixes** [comment]
# Team Analisi del Prodotto

**Support hero:** @Twixes

## Retro

- Some of us had a great offsite
- Others didn't but that's okay
- We got to a bunch of issues in insights

## Hang over items from previous sprint

- Bugs in insights
- Project environments

## OKR

No major change here.

- HogQL-based querying
  - Convert the remaining legacy queries to HogQL and release to public (Thomas, Julian, Marius)
    - 🟢 Insights – they are rolled out!! (still some bugs)
    - ⚪ Cohorts
  - Remove legacy querying backend (Thomas, Julian)
    - 🟠 Clean up or rewrite dashboardLogic 👈 this sprint
    - 🟠 Convert filters to query (insights, notebooks, activity log, experiments)
  - Missing Product Analytics features (Thomas, Julian)
    - ⚪ Breakdowns (multiple) in literally everything
    - 🟠 [Make a list](https://github.com/PostHog/meta/issues/200) based on GitHub issues from customer requests…
    - ⚪ Fix those issues
  - Missing HogQL features (Tom, Marius)
    - 🟠 Type system, JSON 👈 Data Warehouse is on this
    - ⚪ Missing things when building funnels

- Querying and processing performance (Thomas, Julian)
  - Global performance overview dashboards
    - ⚪ Insights
    - ⚪ Exports
    - ⚪ Cohort recalculations
  - Query request tracing
    - ⚪ Possibly query runner Python optimizations
    - ⚪ Exports improvements
  - ⚪ Identify top 5 query optimizations in terms of impact

- Artificial Hog / Post Intelligence (Michael)
  - ⚪ Ask a question to get a magical insight (aware of your taxonomy)
  - ⚪ Figure out infra for upgrading queries and models
  - ⚪ Product-wide framework for opting into sharing with OpenAI

- Activation (side quest: Michael)
  - ⚪ Michael to work with Growth to identify optimizations to getting started with Product Analytics

### High priority

- Fix the most painful bugs reported by users, particularly rooted in new HogQL-based insights (we're on the way down with the pile of open tickets, currently at 90) @all-of-product-analytics
- Clean up the legacy ways of tracking performance 

### Low priority / side quests

- Ensure live events in the activity @Twixes 
- Project environments on hold (also: need to figure out what team should it live in)

**Comment by tiina303** [comment]
Team pipeline (WIP)

- Tomas joining Data warehouse  - collab with Jams on deletes?
- Ted - joining CH team, collab with Tim on performance?

- Xavier and Brett are off this week, so we'll narrow down the details next week.

Items:
- customer.io plugin
- Hog
- PG running out of cpu

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @jurajmajerik     
Days off:
Juraj:  5 days
Phani: 0 days
Dylan: 0 days
Neil: 2 days

## Retro

- Offsite hackathon + wrapup - @jurajmajerik ✅ / @neilkakkar - off sick, this will happen some other time.
- Surveys popup refactor - @jurajmajerik  -- caught up on support instead, lots pending after offsite
- Make sure Dylan's onboarding goes well - @neilkakkar ✅
- Implement the RFC for repeatable surveys - https://github.com/PostHog/meta/pull/203 + some small bugs - @Phanatic 🔄
	- mostly done 
- some excellent survey bugfixes - @dmarticus ✅
	-  https://github.com/PostHog/posthog/pull/22496 and some more in the queue

## Hang over items from previous sprint

---

## OKRs

1. Make sure feature flags can handle 10x current scale
2. Polish new experiments UI & collect feedback
3. Add most requested surveys functionality

### High priority

- Finish up repeated surveys - @Phanatic 
- next start on Event based triggers for surveys - https://github.com/PostHog/product-internal/pull/556 - @Phanatic 
- https://github.com/PostHog/posthog/issues/18102 - @dmarticus 
- NPS survey result fix (ex: [this survey](https://us.posthog.com/project/2/surveys/018b6615-0d7a-0000-bbbc-1ab59bce6866) ) - @dmarticus 
- Surveys popup refactor - @jurajmajerik 
- https://github.com/PostHog/posthog/issues/22131 continue on rewrite - @neilkakkar 

### Low priority / side quests / maybe Neil will get to this next year

- Temporal queues for feature success - @neilkakkar 
- Setup instrumentation for flip-flopping problem of experiment significance - @neilkakkar
